Library/Homebrew/tap.rb

```diff
@@ -280,4 +316,93 @@ def self.each
   def self.names
     map(&:name)
   end
+
+  private
+
+  # @private
+  def formula_file_to_name(file)
+    "#{name}/#{file.basename(".rb")}"
+  end
+
+  # @private
+  def alias_file_to_name(file)
+    "#{name}/#{file.basename}"
+  end
+end
+
+class CoreFormulaTap < Tap
```

Comment:
FYI, here is what I'm planed for future. 
1. We will delete `Library/Formula`, `Library/Aliases` and `Library/Homebrew/formula_rename.rb` in the core.
2. We will export formula files with full git log into a tap repo. In this process, an equivalent  `formula_rename.json` will be created.
3. This core formula tap will be tapped to `Library/Taps/homebrew/homebrew`. At this point, core formula tap is identical to normal tap. Except, it cannot be uninstalled and pinned/unpinned.
4. In `brew.rb`, a migration check will be added to ensure the new tap is installed.

>  If you really, really don't want to use a Module can you remove Tap from the name

At first appearance, module seems to look better. But, its code will be pretty strange as it replies the methods outside formula i.e. from the Tap class to work.

> can you remove Tap from the name (it is not currently a real Tap) e.g. CoreFormulaRepository

I have no objection. But I need to remind you that we won't change this class name even after core formula separation in the future in order to keep backward forward ability. If you and other maintainers are OK this, I will go head to change it to that name.